File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        long revNum = 0;
        boolean sign = x < 0 ? true : false;
        x = Math.abs(x);
        while(x > 0){
            int lastDig = x % 10;
            revNum = revNum * 10 + lastDig;
            x = x / 10;
            if(revNum > Integer.MAX_VALUE){
                return 0;
            }
        }
        return sign == true ? (int)revNum * (-1) : (int) revNum;

    }
}

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        sign = x < 0
        x = abs(x)
        rev_num = 0
        while x > 0:
            last_dig = x % 10
            rev_num = rev_num * 10 + last_dig
            x = x / 10
            if rev_num > ((1 << 31) - 1):
                return 0
        return -rev_num if sign else rev_num
        